view_all_set.php

```diff
@@ -414,26 +425,32 @@
 # 	25: $f_relationship_bug
 # 	26: $f_show_profile
 
-# Set new filter values.  These are stored in a cookie
-$t_view_all_cookie_id = gpc_get_cookie( config_get( 'view_all_cookie' ), '' );
-$t_view_all_cookie = filter_db_get_filter( $t_view_all_cookie_id );
-
-# process the cookie if it exists, it may be blank in a new install
-if( !is_blank( $t_view_all_cookie ) ) {
-	$t_setting_arr = filter_deserialize( $t_view_all_cookie );
-	if( false === $t_setting_arr ) {
-		# couldn't deserialize, if we were trying to use the filter, clear it and reload
-		# for ftype = 0, 1, or 3, we are going to re-write the filter anyways
-		if( !in_array( $f_type, array( 0, 1, 3 ) ) ) {
-			gpc_clear_cookie( 'view_all_cookie' );
-			error_proceed_url( 'view_all_set.php?type=0' );
-			trigger_error( ERROR_FILTER_TOO_OLD, ERROR );
-			exit; # stop here
+if( null !== $f_filter_token ) {
+	# Here the filter is retrieved by the filter gpc parameter
+	$t_setting_arr = current_user_get_bug_filter();
+}
+else {
+	# Set new filter values.  These are stored in a cookie
+	$t_view_all_cookie_id = gpc_get_cookie( config_get( 'view_all_cookie' ), '' );
+	$t_view_all_cookie = filter_db_get_filter( $t_view_all_cookie_id );
+
+	# process the cookie if it exists, it may be blank in a new install
+	if( !is_blank( $t_view_all_cookie ) ) {
+		$t_setting_arr = filter_deserialize( $t_view_all_cookie );
+		if( false === $t_setting_arr ) {
+			# couldn't deserialize, if we were trying to use the filter, clear it and reload
+			# for ftype = 0, 1, or 3, we are going to re-write the filter anyways
+			if( !in_array( $f_type, array( 0, 1, 3 ) ) ) {
+				gpc_clear_cookie( 'view_all_cookie' );
+				error_proceed_url( 'view_all_set.php?type=0' );
+				trigger_error( ERROR_FILTER_TOO_OLD, ERROR );
+				exit; # stop here
+			}
 		}
+	} else {
+		# no cookie found, set it
+		$f_type = 1;
 	}
-} else {
-	# no cookie found, set it
-	$f_type = 1;
```

Comment:
Type 1 is to create a new filter
'temporary' parameter is used so as to save in tokens instead of DB.

What is changed in this page basically is:
Do we have 'temporary' parameter? create new filter, and store it in tokens, not in DB
Do we have 'filter' parameter? retrieve filter from tokens, modify, store in tokens again.